terraform/dev/main.tf

```diff
@@ -0,0 +1,107 @@
+provider "aws" {
+  region = "us-east-1"
+}
+
+##
+# Data providers
+##
+data "aws_vpc" "selected" {
+  id = "${var.vpc_id}"
+}
+
+data "aws_subnet_ids" "app" {
+  vpc_id = "${var.vpc_id}"
+
+  tags {
+    Layer       = "app"
+    stack       = "${var.stack}"
+    application = "${var.app}"
+  }
+}
+
+data "aws_subnet" "app" {
+  count = "${length(data.aws_subnet_ids.app.ids)}"
+  id    = "${data.aws_subnet_ids.app.ids[count.index]}"
+}
+
+data "aws_elb" "elb" {
+  name = "${var.elb_name}"
+}
+
+##
+# Security groups
+##
+resource "aws_security_group" "ci" {
+  name        = "ci-to-app-servers"
+  description = "Allow CI access to app servers"
```

Comment:
Is this necessary?  Does the CI server ever log into the app servers?

Reply:
For the lower environments, particularly `dev`, I'd like CI to be able to deploy code refreshes (via Ansible playbook) on merge to master/develop, bypassing a full AMI build (it takes ~30 mins at the moment).